### DISCUSSION TITLE
Twig and variants

### DISCUSSION THREAD

**Discussion by artisten** [pimcore/pimcore#13108] [Q&A / Support] [answered]
Using Twig to do some admin styling in Pimcore

I can do {% object.getChildren %} and get an array.
But it does not pass objects of type variant.

There is no clear examples of this using Twig that I could find on getting children of type variant.

`{% set elms = object.getChildren([
    \Pimcore\Model\DataObject::OBJECT_TYPE_VARIANT,
    \Pimcore\Model\DataObject::OBJECT_TYPE_OBJECT
]) %}`

`.getChildren([DataObject::OBJECT_TYPE_VARIANT,DataObject::OBJECT_TYPE_OBJECT])`

Any other way to get children of type variant? Or better ways of getting any type of child of object.

Can't find a twig example :)
https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Class_Settings/Variants.html

Trying to collect data from each variant for styling summary information in top parent object using twig.

**Comment by NiklasBr** [comment, marked as answer]
Constants in Twig are not resolved normally, try either `"variant"` directly or `constant('\\Pimcore\\Model\\DataObject::OBJECT_TYPE_VARIANT')`.

**Reply by artisten** [reply]
`.getChildren(['variant'])`

Seems to have worked! Thanks Niklas!

Cheers
